app/src/main/java/ch/hikemate/app/model/elevation/ElevationServiceRepository.kt

```diff
@@ -0,0 +1,133 @@
+package ch.hikemate.app.model.elevation
+
+import ch.hikemate.app.model.route.LatLong
+import java.io.IOException
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.Json.Default.decodeFromString
+import kotlinx.serialization.json.buildJsonArray
+import kotlinx.serialization.json.buildJsonObject
+import kotlinx.serialization.json.put
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.OkHttpClient
+import okhttp3.RequestBody
+
+private const val BASE_URL = "https://api.open-elevation.com/api/v1/lookup"
+private const val LOCATION_KEY = "locations"
+private const val LATITUDE_KEY = "latitude"
+private const val LONGITUDE_KEY = "longitude"
+private const val ACCEPT_HEADER = "Accept"
+private const val APPLICATION_JSON = "application/json"
+private const val CONTENT_TYPE_HEADER = "Content-Type"
+private val JSON_MEDIA_TYPE = "application/json; charset=utf-8".toMediaType()
+
+/** Serializable data class for the response from the Elevation API */
+@Serializable data class ElevationResponse(val results: List<ElevationResult>)
+
+/** Serializable data class for the response from the Elevation API */
+@Serializable
+data class ElevationResult(val latitude: Double, val longitude: Double, val elevation: Double)
+
+/**
+ * A repository for the ElevationService. This class is responsible for making the network request
+ */
+class ElevationServiceRepository(private val client: OkHttpClient) : ElevationService {
+
+  // Cache for the elevation data, indexed by hike ID
+  private val cache = mutableMapOf<Long, List<Double>>()
```

Comment:
Is the cache storing **elevation for all points of the hike** ?
I think we could discuss about how we want to use elevation. We might not need all elevation but 1 out of 4 for example.

Reply:
Yes this is totally correct. This is just a placeholder, since I really don't know how much data could be stored in elevation points. I will open a discussion in telegram.

Reply:
From what was discussed in telegram: we will leave it like this for now and if there is performance/memory issues we'll change it later.